src/Analyzer.PowerShellRuleEngine/PowerShellRuleEngine.cs

```diff
@@ -78,24 +74,34 @@ public IEnumerable<IEvaluation> EvaluateRules(string templateFilePath)
                     {
                         evaluationResults.Add(new PowerShellRuleResult(false, lineNumber));
                     }
-                    evaluations.Add(new PowerShellRuleEvaluation(executionResult.Name, uniqueError.Key, false, evaluationResults));
+                    var ruleDescription = executionResult.Name + ". " + uniqueError.Key;
+                    evaluations.Add(new PowerShellRuleEvaluation(ruleIDs[uniqueError.Key], ruleDescription, false, evaluationResults));
                 }
             }
 
             return evaluations;
         }
 
-        private void AddErrorToDictionary(dynamic error, ref Dictionary<string, SortedSet<int>> uniqueErrors)
+        private void AddErrorToDictionaries(dynamic error, ref Dictionary<string, SortedSet<int>> uniqueErrors, ref Dictionary<string, string> ruleIDs)
```

Comment:
Since these dictionaries are never reassigned to, they don't need to be passed as `ref`.